### PR TITLE
[Example][Improvement] Port AutoMMPredictor to Tabular + Add kaggle house competition example + Improve FT-Transformer example

### PR DESCRIPTION
**Pull request by sxjscience**
*Issue #, if available:*

*Description of changes:*

- Port AutoMMPredictor to Tabular, with the key "AG_AUTOMM_NN".
- Improve the FT-Transformer example
- Add a new example about Kaggle california house price prediction competition.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sxjscience** [comment]
@yiqings

File: tabular/src/autogluon/tabular/models/automm/automm_model.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    @classmethod
+    def _get_default_ag_args(cls) -> dict:
+        default_ag_args = super()._get_default_ag_args()
+        extra_ag_args = {'valid_stacker': False}
+        default_ag_args.update(extra_ag_args)
+        return default_ag_args
+
+    def _set_default_params(self):
+        super()._set_default_params()
+        try_import_autogluon_text()
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:79]: Why here? Seems unnecessary.

File: tabular/src/autogluon/tabular/models/automm/automm_model.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        # Decide name of the label column
+        if 'label' in X.columns:
+            label_col_id = 0
+            while True:
+                self._label_column_name = 'label{}'.format(label_col_id)
+                if self._label_column_name not in X.columns:
+                    break
+                label_col_id += 1
+        else:
+            self._label_column_name = 'label'
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:121]: Maybe worth moving to a dedicated method for better readability

Also: This should be after `self.preprocess` calls, to ensure that the column name wasn't somehow added during preprocessing.

File: tabular/src/autogluon/tabular/models/automm/automm_model.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        X_train = self.preprocess(X, fit=True)
+        if X_val is not None:
+            X_val = self.preprocess(X_val)
+        # Get arguments from kwargs
+        verbosity = kwargs.get('verbosity', 2)
+        num_gpus = kwargs.get('num_gpus', None)
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:127]: why have them in kwargs if you always use them?

File: tabular/src/autogluon/tabular/models/automm/automm_model.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        X_train.insert(len(X_train.columns), self._label_column_name, y)
+        if X_val is not None:
+            X_val.insert(len(X_val.columns), self._label_column_name, y_val)
+
+        verbosity_text = max(0, verbosity - 1)
+        root_logger = logging.getLogger('autogluon')
+        root_log_level = root_logger.level
+        self.model = AutoMMPredictor(label=self._label_column_name,
+                                     problem_type=self.problem_type,
+                                     path=self.path,
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:141]: Is self.path ok? Do we want to add any suffix such as `self.path + 'automm/'`?

File: tabular/src/autogluon/tabular/models/automm/automm_model.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    S_TEXT_NGRAM, S_TEXT_AS_CATEGORY, S_TEXT_SPECIAL, S_IMAGE_PATH
+from autogluon.core.constants import REGRESSION
+from autogluon.core.utils import get_cpu_count, get_gpu_count_torch, try_import_autogluon_text
+from autogluon.core.models import AbstractModel
+
+logger = logging.getLogger(__name__)
+
+
+class AutoMMPredictorModel(AbstractModel):
+    nn_model_name = 'automm_model'
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:20]: Make private?

File: tabular/src/autogluon/tabular/models/automm/automm_model.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        params = self._get_model_params()
+
+        if num_gpus is not None:
+            params['env.num_gpus'] = num_gpus
+
+        self.model.fit(train_data=X_train,
+                       tuning_data=X_val,
+                       time_limit=time_limit,
+                       hyperparameters=params,
+                       seed=params.pop('seed', 0))
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:153]: Do we want to add any post-fit cleanup of disk artifacts?

Comment (Innixma) [inline tabular/src/autogluon/tabular/models/automm/automm_model.py:153]: Why are we popping params here? This is a bit iffy, I'm not even technically sure what happens here, but I assume `hyperparameters` will have the popped version of params. Maybe better to explicitly get the seed popped from params at an earlier point instead of inside the model.fit call for better readability.

**Review by Innixma — COMMENTED** [review]
Looks good! Had some minor comments

File: tabular/src/autogluon/tabular/configs/hyperparameter_configs.py [hunk 7]
```diff
@@ -82,7 +82,7 @@
         'CAT': {},
         'XGB': {},
         # 'FASTAI': {},  # FastAI gets killed if the dataset is large (400K rows).
-        'AG_TEXT_NN': {'presets': 'medium_quality_faster_train'},  # TODO, Support changing the config w.r.t the preset option.
+        'AG_TEXT_NN': {'presets': 'medium_quality_faster_train'},
```
[the remark this answers is not in the document]
Comment (sxjscience) [reply tabular/src/autogluon/tabular/configs/hyperparameter_configs.py:85]: Running benchmarks right now.

**Comment by szha** [comment]
Job PR-1810-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1810/3/index.html

**Comment by szha** [comment]
Job PR-1810-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1810/4/index.html
